### PR TITLE
Stub fixes for mysqli extension

### PR DESCRIPTION
**Pull request by kamil-tekiela**

### CODE REVIEW

File: dictionaries/CallMap.php [hunk 1]
```diff
@@ -8461,25 +8462,25 @@
...
+'mysqli_fetch_array' => ['array|false|null', 'result'=>'mysqli_result', 'mode='=>'int'],
+'mysqli_fetch_assoc' => ['array<string,string>|false|null', 'result'=>'mysqli_result'],
 'mysqli_fetch_column' => ['null|int|float|string|false', 'result'=>'mysqli_result', 'column='=>'int'],
 'mysqli_fetch_field' => ['object|false', 'result'=>'mysqli_result'],
 'mysqli_fetch_field_direct' => ['object|false', 'result'=>'mysqli_result', 'index'=>'int'],
 'mysqli_fetch_fields' => ['array|false', 'result'=>'mysqli_result'],
 'mysqli_fetch_lengths' => ['array|false', 'result'=>'mysqli_result'],
-'mysqli_fetch_object' => ['?object', 'result'=>'mysqli_result', 'class='=>'string', 'constructor_args='=>'?array'],
-'mysqli_fetch_row' => ['?array', 'result'=>'mysqli_result'],
+'mysqli_fetch_object' => ['object|false|null', 'result'=>'mysqli_result', 'class='=>'string', 'constructor_args='=>'?array'],
```
[the remark this answers is not in the document]
Comment (kamil-tekiela) [reply dictionaries/CallMap.php:8472]: Like so?

File: dictionaries/CallMap.php [hunk 2]
```diff
@@ -8513,18 +8514,18 @@
...
-'mysqli_result::fetch_assoc' => ['array<string,string>|null'],
+'mysqli_result::fetch_array' => ['array|false|null', 'mode='=>'int'],
+'mysqli_result::fetch_assoc' => ['array<string,string>|false|null'],
 'mysqli_result::fetch_column' => ['null|int|float|string|false', 'column='=>'int'],
 'mysqli_result::fetch_field' => ['object|false'],
 'mysqli_result::fetch_field_direct' => ['object|false', 'index'=>'int'],
 'mysqli_result::fetch_fields' => ['array|false'],
-'mysqli_result::fetch_object' => ['object|stdClass|null', 'class='=>'string', 'constructor_args='=>'array'],
-'mysqli_result::fetch_row' => ['?array'],
+'mysqli_result::fetch_object' => ['object|false|null', 'class='=>'string', 'constructor_args='=>'array'],
```
[the remark this answers is not in the document]
Comment (weirdan) [reply dictionaries/CallMap.php:8523]:
```suggestion
'mysqli_result::fetch_object' => ['object|false|null', 'class='=>'class-string', 'constructor_args='=>'array'],
```
, for consistency with `mysqli_fetch_object()`

File: stubs/mysqli.phpstub [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+/**
+ * @template TValue
```
Comment (weirdan) [inline stubs/mysqli.phpstub:4]: Extension stubs are loaded conditionally, see https://github.com/vimeo/psalm/blob/5cf660f63b548ccd4a56f62d916ee4d6028e01a3/src/Psalm/Config.php#L1931-L1934

**Review by weirdan — CHANGES_REQUESTED** [review]
Looking great, I've got one last nitpick though.

**Review by weirdan — APPROVED** [review]
:shipit:

**Comment by weirdan** [comment]
Thanks!

**Comment by kamil-tekiela** [comment]
I'm not sure what happened to the docblock in mysqli.phpstub. Maybe I just didn't notice that it is indented.

**Pull request merged**